apiServer/pxone/utils/torpedoutils.go

```diff
@@ -0,0 +1,164 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/portworx/torpedo/drivers/node"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/tests"
+	"math/rand"
+	"net/http"
+)
+
+var (
+	IsTorpedoInitDone bool                 // flag to check if Drivers init is done
+	context           []*scheduler.Context // Application context will be maintained globally for now
+)
+
+// This method checks if test has done InitInstance once or not. If not, we will try to do it.
+func checkTorpedoInit(c *gin.Context) bool {
+	if !IsTorpedoInitDone {
+		InitializeDrivers(c)
+		if !IsTorpedoInitDone {
+			c.JSON(http.StatusInternalServerError, gin.H{
+				"message": "Torpedo Init failed",
+			})
+			return false
+		}
+	}
+	return true
+}
+
+// InitializeDrivers : This API Call will init all Torpedo Drivers. This needs to be run as ginkgo test
+// as multiple ginkgo and gomega dependencies are being called in InitInstance()
+func InitializeDrivers(c *gin.Context) {
+	tests.ParseFlags()
+	tests.InitInstance()
+	IsTorpedoInitDone = true
+}
+
+// GetNodes : This API will return list of all worker nodes in the Cluster
+func GetNodes(c *gin.Context) {
+	if !checkTorpedoInit(c) {
+		return
+	}
+	nodes := node.GetWorkerNodes()
+	c.JSON(http.StatusOK, gin.H{
+		"message": "Nodes are: ",
+		"nodes":   nodes,
+	})
+}
+
+// RebootNode - API Call to Reboot node :
+// pxone/rebootnode/all - Reboots all nodes in cluster
+// pxone/rebootnode/random - Reboots a randomly selected node
+// pxone/rebootnode/<nodename> - Reboots a node with specific node name
+func RebootNode(c *gin.Context) {
+	if !checkTorpedoInit(c) {
+		return
+	}
+	nodename := c.Param("nodename")
+	if nodename == "all" {
+		nodes := node.GetWorkerNodes()
+		for _, n := range nodes {
+			log.InfoD("reboot node: %s", n.Name)
+			err := tests.Inst().N.RebootNode(n, node.RebootNodeOpts{
+				Force: true,
+				ConnectionOpts: node.ConnectionOpts{
+					Timeout:         defaultCommandTimeout,
+					TimeBeforeRetry: defaultCommandRetry,
+				},
+			})
+			if err != nil {
+				c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+				return
+			}
+		}
+		c.JSON(http.StatusOK, gin.H{"message": "All Nodes successfully rebooted"})
+		return
+	} else if nodename == "random" {
```

Comment:
that if ellsif else can we have switch

Reply:
Noted. I will make this to switch-case in the next PR when I add API support for multiple node names